### PR TITLE
Add HI mass functions to the XL pipeline

### PR DESCRIPTION
**Pull request by MatthieuSchaller**
Does what it says in the title.

### CODE REVIEW

**Comment by MatthieuSchaller** [comment]
Also adds a subgrid rho-T 2d histogram. 

Slightly annoyingly this one needs more extended axis than the other ones..

**Comment by JBorrow** [comment]
Would it be helpful to also add these to the COLIBRE config?

**Comment by MatthieuSchaller** [comment]
very much so.

**Comment by JBorrow** [comment]
If you would like to add them now you should duplicate them here (make sure you have the correct axis limits that correspond to the change in resolution, though).

**Comment by MatthieuSchaller** [comment]
Let's leave that for the COLIBRists

File: eagle-xl/config.yml [hunk 1]
```diff
@@ -26,6 +26,11 @@ scripts:
     output_file: density_temperature.png
     section: Density-Temperature
     title: Density-Temperature
+  - filename: scripts/subgrid_density_temperature.py
+    caption: Density-temperature diagram constructed from the sub-grid quantities.
+    output_file: subgrid_density_temperature.png
+    section: Density-Temperature
+    title: Density-Temperature
```
Comment (JBorrow) [inline eagle-xl/config.yml:33]: Should be Subgrid Density-Temperature for the title

**Pull request merged**